src/lib/rules/cloudinary/cloudinary.ts

```diff
@@ -0,0 +1,182 @@
+/**
+ * @fileoverview Image optimization with cloudinary
+ */
+import * as crypto from 'crypto';
+
+import * as getImageData from 'image-size';
+
+import { RuleContext } from '../../rule-context';
+import { IAsyncHTMLElement, IRule, IRuleBuilder, IFetchEnd, IScanEnd } from '../../types';
+import { cutString } from '../../utils/misc';
+import * as logger from '../../utils/logging';
+import { Category } from '../../enums/category';
+
+/* eslint-disable camelcase */
+type cloudinaryResult = {
```

Comment:
I think you should move this to the types folder

Reply:
This is only uses here so I didn't want to pollute the main types folder adding this type. We are doing similar things in other parts like in `src\lib\cli\rules\common.ts`, `tests\helpers\test-server.ts` so that's why I went down this path. Happy to change it if we want to be strict about where all the types should go (but then we should open an issue to track everything)

Reply:
Another option could be to have them in different file in`src/lib/rules/cloudinary/`.